webgpu.h

```diff
@@ -843,6 +838,8 @@ typedef void (*WGPUProcAdapterRequestDevice)(WGPUAdapter adapter, WGPUDeviceDesc
 
 // Procs of Buffer
 typedef void (*WGPUProcBufferDestroy)(WGPUBuffer buffer);
+typedef void const * (*WGPUProcBufferGetConstMappedRange)(WGPUBuffer buffer);
```

Comment:
This is going to be the same function returning the same thing, and C isn't protecting against any writes into this memory. The utility of returning `const *` here in an extra method (that doesn't exist in the upstream API) seems pretty limited.

Reply:
In native we can expose real mappings here, where writes to the mapping have undefined results. By making the pointer const, we avoid a very hidden footgun. Even though the caller can const_cast (or c cast) the pointer, such code would be more immediately suspicious.

Reply:
Are you going to have the same function pointer for both of these signatures? Or is it truly a separate function on your side?

Reply:
It also serve as validation that `GetMappedRange` isn't called for `mapReadAsync`, and any validation error would make the function return nullptr. In Rust you could return something like `Option<&[u8]>` or `Option<&mut [u8]>` (with the appropriate lifetime ofc).

Reply:
Nice race conditions :) The validation specific to `GetMappedRange` would mean it would be two different entrypoints.